Source/DiabloUI/selstart.cpp

```diff
@@ -19,6 +19,14 @@ void ItemSelected(size_t value)
 {
 	auto option = static_cast<StartUpGameMode>(vecDialogItems[value]->m_value);
 	GetOptions().GameMode.gameMode.SetValue(option);
+
+	for (OptionEntryBase *modEntry : GetOptions().Mods.GetEntries()) {
+		if (modEntry->GetName() == "Hellfire") {
+			auto* boolEntry = static_cast<OptionEntryBoolean*>(modEntry);
+			boolEntry->SetValue(option == StartUpGameMode::Hellfire);
+		}
+	}

```

Comment:
Probably should turn this in to a function so it's not repeated, eventually we probably wants a way to enable mods from cli anyway.